src/box/tuple_format.c

```diff
@@ -1176,14 +1176,19 @@ tuple_field_map_create_plain(struct tuple_format *format, const char *tuple,
 	const char *next_pos = pos;
 	for (uint32_t i = 0; i < defined_field_count;
 	     i++, token++, pos = next_pos) {
-		mp_next(&next_pos);
 		field = json_tree_entry(*token, struct tuple_field, token);
-		if (validate) {
-			if (tuple_field_validate(format, field, pos,
-						 next_pos) != 0)
+		bool allow_null = tuple_field_is_nullable(field) ||
+				  tuple_field_has_default(field);
+		if (allow_null && mp_typeof(*pos) == MP_NIL) {
```

Comment:
May be worth using `likely` here.

Reply:
It doesn't affect performance here, and it will make the code harder to understand, because the `else`  branch is actually more likely in most cases.